Taxi/Taxi/Extension/DateExtension.swift

```diff
@@ -0,0 +1,93 @@
+//
+//  DateExtension.swift
+//  Taxi
+//
+//  Created by sanghyo on 2022/06/09.
+//
+
+import Foundation
+
+extension Date {
+    var formatter: DateFormatter {
+        let formatter = DateFormatter()
+        formatter.dateFormat = "yyyy년 MMM"
+        formatter.locale = Locale(identifier: "ko_KR")
+        formatter.timeZone = TimeZone(abbreviation: "KST")
+        return formatter
+    }
+
+    var intFormatter: DateFormatter {
+        let formatter = DateFormatter()
+        formatter.dateFormat = "yyyyMMdd"
+        formatter.locale = Locale(identifier: "ko_KR")
+        formatter.timeZone = TimeZone(abbreviation: "KST")
+        return formatter
+    }
+
+    var dateComponents: DateComponents {
+        return Calendar.current.dateComponents([.year, .month, .day], from: self)
+    }
+
+    var day: Int {
+        return self.dateComponents.day ?? -1
+    }
+
+    var month: Int {
+        return self.dateComponents.month ?? -1
+    }
+
+    var year: Int {
+        return self.dateComponents.year ?? -1
+    }
+
+    var formattedString: String {
+        return self.formatter.string(from: self)
+    }
+
+    /// date를 yyyyMMdd의 형태로 바꿈
+    var formattedInt: Int {
+        let formattedStr = self.intFormatter.string(from: self)
+        return Int(formattedStr) ?? -1
+    }
+
+    func monthlyDayCount() -> Int {
+        let calendar = Calendar.current
+        let range = calendar.range(of: .day, in: .month, for: self)
+        return range?.count ?? 30
+    }
+
+    func getMonthDates() -> [Date] {
+        let calendar = Calendar.current
+        guard let startDate = calendar.date(
+            from: calendar.dateComponents(
+                [.year, .month],
+                from: self
+            )
+        )else { return [] }
+
+        guard let range = calendar.range(of: .day, in: .month, for: startDate)
+        else { return [] }
+
+        return range.compactMap { day -> Date in
+            calendar.date(byAdding: .day, value: day - 1, to: startDate) ?? Date()
+        }
+    }
+
+    func isSameDay(_ comparedDate: Date) -> Bool {
+        let calendar = Calendar.current
+        return calendar.isDate(self, inSameDayAs: comparedDate)
+    }
+
+    func isToday() -> Bool {
+        let calendar = Calendar.current
+        return calendar.isDate(self, inSameDayAs: Date())
+    }
```

Comment:
```suggestion
    var isToday: Bool {
        let calendar = Calendar.current
        return calendar.isDate(self, inSameDayAs: Date())
    }
```

computed property로도 될 수 있을 것 같아요 ~~
`class Calendar`의 extension으로 되는게 적절해보이기도 하네요

Reply:
21a1f4fb0bb31a36e708f2da98acdafc35f5f5e0 반영 완료하였습니다.
인자를 받지 않는 함수는 computed property로 대체될 수 있으며, 동사형의 명명법은 피하는게 좋다!